src/command_modules/azure-cli-iot/azure/cli/command_modules/iot/custom.py

```diff
@@ -44,8 +44,42 @@ def iot_hub_create(client, name, resource_group_name,
     return result
 
 
+def iot_hub_get_connection_string(client, name, resource_group_name, policy_name='iothubowner'):
+    access_policies = client.list_keys(resource_group_name, name).next()
+    if access_policies is None:
+        raise CLIError('No policy found from IoT Hub: {}.'.format(name))
+    logger.info('Shared Access Polices: %s', str(access_policies))
+    try:
+        access_policy = next(x for x in access_policies
+                             if policy_name.lower() == x.key_name.lower())
+    except StopIteration:
+        raise CLIError('No policy found with name {}'.format(policy_name))
+
+    connection_string = 'HostName={}.azure-devices.net;SharedAccessKeyName={};SharedAccessKey={}'.format(name,
+                                                                                                         policy_name,
+                                                                                                         access_policy.primary_key)
+    return {'connectionString': connection_string}
+
+
 def iot_device_create(client, resource_group_name, hub, device_id):
```

Comment:
Regarding parameter naming, I'd have to discuss with our PM. Will ping you later when we have a conclusion.